content/automate/ManageIQ/Transformation/Ansible.class/__methods__/checkplaybookasaservice.rb

```diff
@@ -0,0 +1,51 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Ansible
+        class CheckPlaybookAsAService
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def set_retry(message, interval = '1.minutes')
+          end
+
+          def main
+            transformation_hook = @handle.inputs['transformation_hook']
+            task = @handle.root['service_template_transformation_plan_task']
+            service_request_id = task.get_option("#{transformation_hook}_ansible_playbook_service_request_id")
+
+            if service_request_id.present?
+              service_request = @handle.vmdb(:miq_request).find_by(:id => service_request_id)
+
+              playbooks_status = task.get_option(:playbooks) || {}
+              playbooks_status[transformation_hook] = { :job_state => service_request.request_state }
+
+              if service_request.request_state == 'finished'
+                @handle.log(:info, "Ansible playbook service request (id: #{service_request_id}) is finished.")
+                playbooks_status[transformation_hook][:job_status] = service_request.status
+                playbooks_status[transformation_hook][:job_id] = service_request.miq_request_tasks.first.destination.service_resources.first.resource.id
+                task.set_option(:playbooks, playbooks_status)
+                if service_request.status == 'Error' && transformation_hook == 'pre'
```

Comment:
@fdupont-redhat is it intentional to ignore the errors for post playbooks?

Reply:
@mkanoor 
Yes, it is. We consider that the migration has gone far enough to not cancel it for a post migration playbook failure. The playbook or any other problem can be fixed afterwards, with a warning message. The UI knows that the playbook has failed, and it can show the log to the user.